source/Sashimi.Azure/Web/AzureEnvironmentsListAction.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Octopus.Server.Extensibility.Extensions.Infrastructure.Web.Api;
+
+namespace Sashimi.Azure.Web
+{
+    public class AzureEnvironmentsListAction : IAsyncApiAction
+    {
+        public Task ExecuteAsync(OctoContext context)
+        {
+            context.Response.AsOctopusJson(GetEnvironments());
```

Comment:
We could call `GetEnvironments()` in a static ctor and then just serve that in `ExecuteAsync`

Reply:
We could, then we need to make this action singleton instead. Otherwise, there is no point of doing it.

Reply:
yes static field and static ctor